api/handlers/listhandler.py

```diff
@@ -576,3 +584,132 @@ def packfile_end(self, cont_name, **kwargs):
+        container, permchecker, storage, mongo_validator, _, keycheck = self._initialize_request(cont_name, list_name, _id)
+
+        permchecker(noop)('POST', _id=_id)
+        payload = upload.process_upload(self.request, upload.Strategy.analysis, origin=self.origin)
+        payload['created'] = datetime.datetime.utcnow()
+        payload['user'] = payload.get('user', self.uid)
+        result = keycheck(mongo_validator(storage.exec_op))('POST', _id=_id, payload=payload)
+        if result.modified_count == 1:
+            return {'_id': payload['_id']}
+        else:
+            self.abort(404, 'Element not added in list {} of container {} {}'.format(storage.list_name, storage.cont_name, _id))
+
+    def download(self, cont_name, list_name, **kwargs):
+        _id = kwargs.pop('cid')
+        container, permchecker, storage, _, _, _ = self._initialize_request(cont_name, list_name, _id)
+        filename = kwargs.get('name')
+        ticket_id = self.get_param('ticket')
+        if not ticket_id:
+            permchecker(noop)('GET', _id=_id)
+        analysis_id = kwargs.get('_id')
+        fileinfo = [f['files'] for f in storage.get_fileinfo(_id, analysis_id, filename)]
+        if not ticket_id:
+            if filename:
+                total_size = fileinfo[0]['size']
+                file_cnt = 1
+                ticket = util.download_ticket(self.request.client_addr, 'file', _id, filename, total_size)
+            else:
+                targets, total_size, file_cnt = self._prepare_batch(fileinfo)
+                filename = 'analysis_' + analysis_id + '.tar'
+                ticket = util.download_ticket(self.request.client_addr, 'batch', targets, filename, total_size)
+            return {
+                'ticket': config.db.downloads.insert_one(ticket).inserted_id,
+                'size': total_size,
+                'file_cnt': file_cnt,
+                'filename': filename
+            }
+        else:
+            ticket = self._check_ticket(ticket_id, _id, filename)
+            if not filename:
+                self._send_batch(ticket)
+                return
+            if not fileinfo:
+                self.abort(404, '{} doesn''t exist'.format(filename))
+            fileinfo = fileinfo[0]
+            filepath = os.path.join(
+                config.get_item('persistent', 'data_path'),
+                util.path_from_hash(fileinfo['hash'])
+            )
+            filename = fileinfo['name']
+            self.response.app_iter = open(filepath, 'rb')
+            self.response.headers['Content-Length'] = str(fileinfo['size']) # must be set after setting app_iter
+            if self.is_true('view'):
+                self.response.headers['Content-Type'] = str(fileinfo.get('mimetype', 'application/octet-stream'))
+            else:
+                self.response.headers['Content-Type'] = 'application/octet-stream'
+                self.response.headers['Content-Disposition'] = 'attachment; filename=' + str(filename)
```

Comment:
Is the download behaviour here different than other containers? If so, can you explain & document?